File: server/src/protocol/MotorStatusCommandBuilder.cpp
#include "MotorStatusBuilder.h"
#include "MotorStatusCommand.h"

using namespace std;
using namespace communication;

MotorStatusBuilder::MotorStatusBuilder()
{}

MotorStatusBuilder::~MotorStatusBuilder()
{}

unique_ptr<Command> MotorStatusBuilder::create(const vector<uint8_t> &commandInBytes)
{
    auto command = make_unique<MotorStatusCommand>();

    return move(command);
}

File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/MotorStatusCommandBuilder.java
package wpam.mobile_client.protocol;

import java.io.Closeable;
import java.util.ArrayList;

public final class MotorStatusCommandBuilder implements CommandBuilder, Closeable
{
	public MotorStatusCommandBuilder()
	{
	}
	public final void close()
	{
	}

	@Override
	public Command create(ArrayList<Integer> commandInBytes)
	{
		Command command  = new MotorStatusCommand();
		return command;
	}
}